RecoBTag/SoftLepton/interface/ElectronTagger.h

```diff
@@ -17,22 +17,33 @@ class ElectronTagger : public JetTagComputer {
 public:
 
   /// explicit ctor 
-  explicit ElectronTagger(const edm::ParameterSet & configuration) : 
-    m_selector(configuration)
-  { 
-    uses("seTagInfos"); 
-  }
-  
-  /// dtor
-  virtual ~ElectronTagger() { }
-
-  /// b-tag a jet based on track-to-jet parameters in the extened info collection
+ ElectronTagger(const edm::ParameterSet & );
+ ~ElectronTagger(); 
   virtual float discriminator(const TagInfoHelper & tagInfo) const;
-
+//  std::vector<string> vecstr;
+//  string path_mvaWeightFileEleID;
 private:
-
   btag::LeptonSelector m_selector;
-
+  TRandom3* random;
+  edm::FileInPath WeightFile;
+  MvaSoftEleEstimator* mvaID;
 };
 
-#endif // RecoBTag_SoftLepton_ElectronTagger_h
+ElectronTagger::ElectronTagger(const edm::ParameterSet & configuration):
+    m_selector(configuration)
+  {
+	uses("seTagInfos");
+        random=new TRandom3();
```

Comment:
I don't see this random used anywhere.
There is a local TRandom3 created in .cc, maybe switch to using this one (removes new-delete cycle)?
(make sure to set the seed as before/now, in a deterministic way)